tutorial/advanced/t12_hyperparameter_search.ipynb

```diff
@@ -0,0 +1,676 @@
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "best search result:\n",
+      "({'x': 2.9, 'search_idx': 3}, 0.010000000000000018)\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(\"best search result:\")\n",
+    "print(grid_search.get_best_results())"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "search history:\n",
+      "[({'x': 0.5, 'search_idx': 1}, 6.25), ({'x': 1.5, 'search_idx': 2}, 2.25), ({'x': 2.9, 'search_idx': 3}, 0.010000000000000018), ({'x': 4, 'search_idx': 4}, 1), ({'x': 5.3, 'search_idx': 5}, 5.289999999999999)]\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(\"search history:\")\n",
+    "print(grid_search.get_search_results())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<a id='saveload'></a>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Saving and loading search results\n",
+    "\n",
+    "Once the search is done, you can also save the search results into the disk and later load it back using `save` and `load` method:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "FastEstimator-Search: Saving the search state to /var/folders/cd/9k2rks597yl99yttyxmhkw7h0000gn/T/tmp6ekqb93e/grid_search.json\n",
+      "FastEstimator-Search: Loading the search state from /var/folders/cd/9k2rks597yl99yttyxmhkw7h0000gn/T/tmp6ekqb93e/grid_search.json\n",
+      "[{'x': 2.9, 'search_idx': 3}, 0.010000000000000018]\n"
+     ]
+    }
+   ],
+   "source": [
+    "import tempfile\n",
+    "save_dir = tempfile.mkdtemp()\n",
+    "\n",
+    "# save the state to save_dir\n",
+    "grid_search.save(save_dir) \n",
+    "\n",
+    "# instantiate a new object\n",
+    "grid_search2 = GridSearch(score_fn=objective_fn, params={\"x\": [0.5, 1.5, 2.9, 4, 5.3]}, best_mode=\"min\") \n",
+    "\n",
+    "# load the previously saved state\n",
+    "grid_search2.load(save_dir)\n",
+    "\n",
+    "# display the best result of the loaded instance\n",
+    "print(grid_search2.get_best_results()) "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<a id='interruption'></a>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Interruption-resilient search\n",
+    "When you run search on a hardware that can be interrupted (like spot instance), you can provide a `save_dir` argument when calling `fit`. As a result, the search will automatically back up its result after each evaluation. Furthermore, when calling the `fit` using the same `save_dir` the second time, it will first load the search results and pick up from where it left off. \n",
```

Comment:
(like **an AWS** spot instance)

Furthermore, when **calling `fit`** using the same

search results and **then** pick up from where

Reply:
n